src/compas_ags/diagrams/forcediagram.py

```diff
@@ -301,27 +319,50 @@ def constraints_from_dual(self, tol=10e-4):
         ForceDiagram is modified in place.
         """
         edge_index = self.dual.edge_index()
+        _edges = list(self.edges())
```

Comment:
why is this underscored?

Reply:
Because they are edges of the force diagram. But I agree that this is not really needed. I will modify